shell/platform/common/cpp/text_input_model.cc

```diff
@@ -172,4 +214,13 @@ std::string TextInputModel::GetText() const {
   return utf8_converter.to_bytes(text_);
 }
 
+int TextInputModel::GetCursorOffset() const {
+  // Measure the length of the current text up to the cursor.
+  // There is probably a much more efficient way of doing this.
```

Comment:
(I think you could walk each character and determine its UTF-8 byte count based what range the UTF-16 code unit falls in, but I'm not 100% sure since I don't know if the UTF-16->UTF-8 conversion used in GetText would be guaranteed to generate the same UTF-8 representations, vs. doing some kind of normalization.)

Reply:
I decided the current implementation is probably the safest and should only be optimised if we need to later.